common/src/main/java/com/wynntils/gui/screens/WynntilsScreen.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.gui.screens;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.utils.McUtils;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+
+public abstract class WynntilsScreen extends Screen {
+    protected WynntilsScreen(Component component) {
+        super(component);
+    }
+
+    private void failure(String method, Throwable e) {
+        WynntilsMod.error("Failure in " + this.getClass().getSimpleName() + "." + method + "()", e);
+        McUtils.sendMessageToClient(Component.literal("Wynntils: Failure in " + method + " in "
+                        + this.getClass().getSimpleName() + ". Screen forcefully closed.")
+                .withStyle(ChatFormatting.RED));
+        McUtils.mc().setScreen(null);
+    }
+
+    @Override
+    protected final void init() {
+        try {
+            super.init();
+            doInit();
+        } catch (Throwable t) {
+            failure("init", t);
+        }
+    }
+
+    protected abstract void doInit();
+
+    @Override
+    public final void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        try {
+            super.render(poseStack, mouseX, mouseY, partialTick);
```

Comment:
Why? Can't we decide in doRender if we want to call super? Can we even do that with this change? Since super.render will be this call, not `Screen#render`?

Reply:
Not every render call in screens call super, some of them do renders.

Reply:
To be able to call super.render() from the subclasses we would need to provide a pass-through, like `public void superRender(PoseStack poseStack, ...) { super.render(...) }`.

But I think a failure to call super.render is indicative of a design/inheritance problem.

Reply:
But that means this is not a great solution?

Reply:
Yes, sure, I'll give you that. This is not a great solution. But it is *a* solution. And it won't crash.

Reply:
Hm, I realize therein lies the answer to your problem. I solved the issue half-way; I changed calls from super.render to super.doRender. But, of course, I should add a call to super.render in the top level doRender. Then everything will be just fine.

Reply:
Or... at least I could have sworn I changed those call. I can't find that in the PR though. I must have missed it.

Reply:
Is this acceptable now?